src/librustdoc/html/static/js/settings.js

```diff
@@ -246,25 +268,36 @@
     } else {
         // We replace the existing "onclick" callback.
         const settingsButton = getSettingsButton();
+        const settingsMenu = document.getElementById("settings");
+        window.hideSettings = function() {
+            // switchDisplayedElement(null);
+            settingsMenu.style.display = "none";
+        };
         settingsButton.onclick = function(event) {
+            if (checkIfElemIsInParents(event.target, settingsMenu)) {
+                return;
+            }
             event.preventDefault();
-            if (settingsMenu.parentElement.id === NOT_DISPLAYED_ID) {
-                switchDisplayedElement(settingsMenu);
-            } else {
+            if (settingsMenu.style.display !== "none") {
                 window.hideSettings();
+            } else {
+                displaySettings();
             }
         };
-        window.hideSettings = function() {
-            switchDisplayedElement(null);
-        };
+        settingsButton.onblur = blurHandler;
+        settingsButton.querySelector("a").onblur = blurHandler;
+        onEachLazy(settingsMenu.querySelectorAll("input"), e => {
+            e.onblur = blurHandler;
```

Comment:
I think we usually use `el` to name elements (`e` is usually an Event).

Reply:
Good point!